pkg/controllers/managementuser/nodesyncer/nodessyncer.go

```diff
@@ -822,6 +832,25 @@ func (m *nodesSyncer) isClusterRestoring() (bool, error) {
 		cluster.Spec.RancherKubernetesEngineConfig.Restore.Restore {
 		return true, nil
 	}
+	if strings.HasPrefix(cluster.Name, "c-m-") {
+		provCluster, err := m.provClusterCache.Get(cluster.Spec.FleetWorkspaceName, cluster.Spec.DisplayName)
+		if err != nil {
+			return false, err
+		}
+		capiCluster, err := m.capiClusterCache.Get(provCluster.Namespace, provCluster.Name)
+		if err != nil {
+			return false, err
+		}
+		if capiCluster.Spec.ControlPlaneRef.Kind != "RKEControlPlane" || capiCluster.Spec.ControlPlaneRef.APIVersion != "rke.cattle.io/v1" {
+			return false, nil
+		}
+		controlplane, err := m.rkeControlPlaneCache.Get(capiCluster.Spec.ControlPlaneRef.Namespace, capiCluster.Spec.ControlPlaneRef.Name)
+		if err != nil {
+			return false, err
+		}
+		return controlplane.Status.ETCDSnapshotRestorePhase != "" && controlplane.Status.ETCDSnapshotRestorePhase != rkev1.ETCDSnapshotPhaseFinished, nil
```

Comment:
there is also the case where `EtcdSnapshotRestorePhase` is `Failed`

Reply:
I figured there wasn't much of a point of syncing anything with the cluster if it was failed, I can update though.

Reply:
I think previous versions of rancher were setting that to `failed` immediately as a "default" value